pkg/cni_pb/cni.proto

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2019 Red Hat and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+// Package cni provides remote CNI functionality over gRPC. It follows
+// the CNI specification at https://github.com/containernetworking/cni/blob/master/SPEC.md
+package cni;
+
+// The remote CNI service definition.
+service RemoteCNI {
+  // The request to add a container to network.
+  rpc Add (CNIRequest) returns (CNIReply) {}
+
+  // The request to delete a container from network.
+  rpc Delete (CNIRequest) returns (CNIReply) {}
+}
+
+// The request to add a container to network. Corresponds to the CNI specification
+// at https://github.com/containernetworking/cni/blob/master/SPEC.md#parameters
+message CNIRequest {
+  // The version of CNI spec that the caller is using (container management system or the invoking plugin).
+  string version = 1;
+
+  // Container ID unique across an administrative domain while the container is live.
+  string container_id = 2;
+
+  // Path to the network namespace to be added, i.e. /proc/[pid]/ns/net or a bind-mount/link to it.
+  string network_namespace = 3;
+
+  // Name of the interface inside the container. This is the name that should be assigned to the interface created inside the container (network namespace); consequently it must comply with the standard Linux restrictions on interface names.
+  string interface_name = 4;
+
+  // Network configuration. This is a JSON document describing a network to which a container can be joined.
+  message NetworkConfig {
+    // version of CNI specification
+    string cniVersion = 1;
+
+    // Network name which is unique across all containers on the host
+    string name = 2;
+
+    // Filename of CNI plugin executable
+    string type = 3;
+
+    // Additional dictionary arguments provided by container runtime (optional)
+    map<string, string> args = 4;
+
+    // Sets up IP masquerade on the host for this network. Optional.
+    bool ipMasq = 5;
+
+    CNIReply.DNS dns = 6;
+  }
+
+  NetworkConfig network_config = 5;
+
+  // Extra arguments passed to CNI plugin. Optional.
+  string extra_arguments = 6;
```

Comment:
should extra_arguments be a map?

Reply:
yes it should. I'll change that and going to move this PR into the agent and store the protobuf there as we talked about instead of in the client side.